sql/mdl.h

```diff
@@ -1102,7 +1102,7 @@ class MDL_context
   /**
     Ticket hash. Stores only locked tickets.
   */
-  open_address_hash<MDL_key_trait<MDL_ticket>, hash_trait<MDL_ticket, MDL_key_trait<MDL_ticket> > > ticket_hash;
+  open_address_hash<MDL_key_trait<MDL_ticket>, hash_trait<MDL_ticket,MDL_key> > ticket_hash;
```

Comment:
This only makes it trickier to understand. How do you imagine a developer who wants just an `std::set<Key, Value>` analogue use this?